processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -341,7 +435,18 @@ public void advance()
                       if (Thread.interrupted()) {
                         throw new QueryInterruptedException(new InterruptedException());
                       }
-                      cursorOffset.increment();
+                      cursorOffsetHolder.get().increment();
+
+                      while (!isDone()) {
```

Comment:
current results for reading 1,500,000 rows from a long column, no filters:
```
master
FilterPartitionBenchmark.longRead           1500000     basic  avgt   50  44587.905 ± 116.094  us/op

patch
FilterPartitionBenchmark.longRead           1500000     basic  avgt   50  44774.780 ± 117.027  us/op
```

I'll rerun the query benchmarks as well with recent changes.